### PR TITLE
Loger(

### PR DESCRIPTION
**Pull request by BogdanGusak1993**
Зробив щось схоже на логер( 
Чи можна було використовувати ILoger у цій задачі?

### CODE REVIEW

File: LogerHW5/LogerHW5/Loger.cs
Comment (petrochernykh) [inline LogerHW5/LogerHW5/Loger.cs:1]: Interesting decision to use struct but it's better to use static class for Logger. As generally we need one instance of the Logger per app lietime

File: LogerHW5/LogerHW5/Result.cs [hunk 1]
```diff
@@ -0,0 +1,26 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace LogerHW5
+{
+    internal class Result : Action
```
Comment (petrochernykh) [inline LogerHW5/LogerHW5/Result.cs:9]: Result mustn't be Child of Action this is is-a relationship, but we need has-a where Action will return Result

File: LogerHW5/LogerHW5/Loger.cs [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+            Console.WriteLine($"{DateTime.Now}:{_logType}:{_logtext}");
+
+            _allLog = $"{_allLog}"+$"\n{_logtext}";
+        }
+        public void GetLogText()
+        {
+            _fileWay = "C:\\Users\\User\\Desktop\\Log.txt";
+
+            File.WriteAllText(_fileWay, _allLog);
```
Comment (petrochernykh) [inline LogerHW5/LogerHW5/Loger.cs:32]: Tricky moment, if I don't have such a class, e will get an exception

File: LogerHW5/LogerHW5/Program.cs [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+                    {
+                        result = result.SkippedLogic();
+                    }
+                    else
+                    {
+                        result = result.BrokeLogic();
+                        loger.ShowLog("Action faild by a resone: random goes 3");
+                    }
+                }
+                loger.GetLogText();
```
Comment (petrochernykh) [inline LogerHW5/LogerHW5/Program.cs:34]: It's better to move this logic inside some class (ProcessLoggerService smth like this)

File: LogerHW5/LogerHW5/Loger.cs
Comment (BogdanGusak1993) [reply LogerHW5/LogerHW5/Loger.cs:1]: Ye agree struct was a bit unnecessary here.

File: LogerHW5/LogerHW5/Loger.cs (hunk 2 above)
Comment (BogdanGusak1993) [reply LogerHW5/LogerHW5/Loger.cs:32]: Oh i see now.

File: LogerHW5/LogerHW5/Program.cs (hunk 3 above)
Comment (BogdanGusak1993) [reply LogerHW5/LogerHW5/Program.cs:34]: Thought it`s unnecessary here, because we don`t use this code more then once.

**Pull request merged**